src/PathMappingRepository.php

```diff
@@ -80,18 +81,54 @@ public function find($query, $language = 'glob')
     public function contains($query, $language = 'glob')
     {
         $this->validateSearchLanguage($language);
+        $query = $this->sanitizePath($query);
 
-        return $this->search($this->sanitizePath($query), true)->count() > 0;
+        return !$this->search($query, true)->isEmpty();
     }
 
     /**
-     * Not supported.
-     *
      * {@inheritdoc}
      */
     public function remove($query, $language = 'glob')
     {
-        throw new BadMethodCallException(sprintf('remove\(\) is not supported in %s', __CLASS__));
+        $query = $this->sanitizePath($query);
+
+        Assert::notEq('', trim($query, '/'), 'The root directory cannot be removed.');
```

Comment:
In this PR? I think we should create another PR to do that everywhere needed, it doesn't seem very related to this.